### PR TITLE
Use AutoConfiguration.imports for all auto-configs

### PR DESCRIPTION
**Pull request by onobc**

### CODE REVIEW

File: spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/ApplicationConfigurationMetadataResolverAutoConfiguration.java [hunk 1]
```diff
@@ -29,6 +30,7 @@
  * @author Eric Bottard
  * @author Christian Tzolov
  */
+@AutoConfiguration
 @Configuration
```
Comment (cppwfs) [inline spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/ApplicationConfigurationMetadataResolverAutoConfiguration.java:34]: Should this configuration be removed?
`@AutoConfiguration`'s definition  contains `@Configuration(proxyBeanMethods = false)`.
Unless there is a reason to maintain it.   
I have seen this a couple times in the PR.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java [hunk 2]
```diff
@@ -141,7 +142,6 @@ public StreamBuilder streamBuilder(DataFlowOperations dataFlowOperations){
 	}
 
 	@ConditionalOnProperty(prefix = DataFlowPropertyKeys.PREFIX + "client.authentication", name = "client-id")
-	@Configuration
```
Comment (cppwfs) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java:144]: Just curious if this removal was necessary?   Seems like this should not have been removed.   Thoughts?

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java [hunk 3]
```diff
@@ -223,7 +223,6 @@ public RestControllerAdvice restControllerAdvice() {
 
 
 
-	@Configuration
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java:226]: Again, I may be incorrect.   Probably am...   But should we remove `@Configuration` for these static classes?

**Review by cppwfs — COMMENTED** [review]
Looks great, just a couple of nitpicks.   Thanks for digging into this!

File: spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/ApplicationConfigurationMetadataResolverAutoConfiguration.java (hunk 1 above)
Comment (onobc) [reply spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/ApplicationConfigurationMetadataResolverAutoConfiguration.java:34]: Yep. Not sure why it was left in by OpenRewrite. I'll yank. Good catch.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java (hunk 2 above)
Comment (onobc) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java:144]: This is a Spring Framework 6.x (maybe earlier) best practice. Only top-level config needs to be registered - nested configurations are automatically registered. Look for `With nested @Configuration` in the [@Configuration javadocs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Configuration.html) for more details.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java (hunk 3 above)
Comment (onobc) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java:226]: Ditto https://github.com/spring-cloud/spring-cloud-dataflow/pull/5668/files/7919ace73a567e942b9e0f86bbfa2fca8ca78e27#r1483619438

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java (hunk 2 above)
Comment (cppwfs) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java:144]: But when I read it, it still shows the `@Configuration` added to the inner static class in the sample  and in the doc, it says it means we don't have to add the `@Import` for the static inner class.   
Or is Glenn just being stubborn.   :-)

Comment (onobc) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java:144]: Ahh yeh, that is not the proper doc for this. Let me back up my comment or we can simply remove this. It was put in by OpenRewrite.

Comment (cppwfs) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java:144]: I was just making sure rewrite didn't take out the wrong `@Configuration`.

Comment (onobc) [reply spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java:144]: Updated - added back in.

**Comment by cppwfs** [comment]
LGTM Thanks for doing this!

**Comment by cppwfs** [comment]
LGTM!    Thanks for doing this! Squashed and Merged

**Pull request closed without merging**